parquet-thrift/src/main/java/parquet/thrift/struct/ThriftType.java

```diff
@@ -175,9 +180,35 @@ final public void visit(StringType stringType) {
 
     private final ThriftField[] childById;
 
+  /**
+   * Whether a struct is a union is not always known, because it was not always
+   * written to the metadata files.
+   *
+   * We should always know this in the write path, but may not in the read path.
+   */
+   public enum IsUnion {
+      YES,
+      NO,
+      UNKNOWN
+   }
```

Comment:
StructType is already used (this is a member of class StructType), having trouble coming up with a reasonable name... `StructKind`? seems like a different way of saying StructType and doesn't make it clear what it's for.  `StructSubType`? I went with `IsUnion` because it's there only to explain whether it's a union or not. `StructOrUnion`? I'd like Union to be in the name so that readers see that it's basically here to describe "is this struct a union?"